plugin/session_view.py

```diff
@@ -75,6 +84,63 @@ def __del__(self) -> None:
         for severity in range(1, len(DIAGNOSTIC_SEVERITY) + 1):
             self.view.erase_regions(self.diagnostics_key(severity))
 
+    def _clear_auto_complete_triggers(self, settings: sublime.Settings) -> None:
+        '''Remove all of our modifications to the view's "auto_complete_triggers"'''
+        triggers = settings.get(self.AC_TRIGGERS_KEY)
+        if isinstance(triggers, list):
+            triggers = [t for t in triggers if self.session.config.name != t.get("server", "")]
+            settings.set(self.AC_TRIGGERS_KEY, triggers)
+
+    def _setup_auto_complete_triggers(self, settings: sublime.Settings) -> None:
+        """Register trigger characters from static capabilities of the server."""
+        trigger_chars = self.session.get_capability(self.TRIGGER_CHARACTERS_KEY)
+        if isinstance(trigger_chars, list):
+            self._apply_auto_complete_triggers(settings, trigger_chars)
+
+    def _register_auto_complete_triggers(self, registration_id: str, trigger_chars: List[str]) -> None:
+        """Register trigger characters from a dynamic server registration."""
+        self._apply_auto_complete_triggers(self.view.settings(), trigger_chars, registration_id)
+
+    def _unregister_auto_complete_triggers(self, registration_id: str) -> None:
+        """Remove trigger characters that were previously dynamically registered."""
+        settings = self.view.settings()
+        triggers = settings.get(self.AC_TRIGGERS_KEY)
+        if isinstance(triggers, list):
+            new_triggers = []  # type: List[Dict[str, str]]
+            name = self.session.config.name
+            for trigger in triggers:
+                if trigger.get("server", "") == name and trigger.get("registration_id", "") == registration_id:
+                    continue
+                new_triggers.append(trigger)
+            settings.set(self.AC_TRIGGERS_KEY, triggers)
+
+    def _apply_auto_complete_triggers(
+        self,
+        settings: sublime.Settings,
+        trigger_chars: List[str],
+        registration_id: Optional[str] = None
+    ) -> None:
+        """This method actually modifies the auto_complete_triggers entries for the view."""
+        selector = self.session.config.auto_complete_selector
+        if not selector:
+            # If the user did not set up an auto_complete_selector for this server configuration, fallback to the
+            # "global" auto_complete_selector of the view.
+            selector = str(settings.get("auto_complete_selector"))
```

Comment:
So setting this one is not equivalent to omitting it completely?

Reply:
I'm not certain we need this.

Reply:
Yes it at least bounds the trigger chars to whatever the global ac selector is. Previously, trigger chars would trigger in comments as well.